P2PEnabledConference.js

```diff
@@ -0,0 +1,886 @@
+            },
+            (error) => {
+                logger.error(
+                    "Attach " + localTracks + " to JVB failed!", error);
+            });
+    }
+
+    /**
+     * Adds remote tracks to the conference associated with the P2P session.
+     * @private
+     */
+    _addP2PRemoteTracks () {
+        this._addRemoteTracks("P2P", this.p2pJingleSession);
+    }
+
+    /**
+     * Adds remote tracks to the conference associated with the JVB session.
+     * @private
+     */
+    _addRemoteJVBTracks () {
+        this._addRemoteTracks("JVB", this.jvbJingleSession);
+    }
+
+    /**
+     * Generates fake "remote track added" events for given Jingle session.
+     * @param {string} logName the session's nickname which will appear in log
+     * messages.
+     * @param {JingleSessionPC} jingleSession the session for which remote
+     * tracks will be added.
+     * @private
+     */
+    _addRemoteTracks (logName, jingleSession) {
+        if (!jingleSession) {
+            logger.info(
+                "Not adding remote " + logName + " tracks - no session yet");
+            return;
+        }
+        const remoteTracks = jingleSession.peerconnection.getRemoteTracks();
+        remoteTracks.forEach(
+            (track) => {
+                logger.info("Adding remote " + logName + " track: " + track);
+                this.rtc.eventEmitter.emit(
+                    RTCEvents.REMOTE_TRACK_ADDED, track);
+            });
+    }
+
+    /**
+     * Creates fake {@link ChatRoom} which is to be used by the P2P Jingle
+     * Session.
+     * @param {boolean} isInitiator indicates whether the room is to be create
+     * for 'initiator' or 'responder'
+     * @return {FakeChatRoomLayer}
+     * @private
+     */
+    _createFakeRoom(isInitiator) {
+        return new FakeChatRoomLayer(this, isInitiator);
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    _doReplaceTrack (oldTrack, newTrack) {
+        const allPromises = [super._doReplaceTrack(oldTrack, newTrack)];
+        if (this.p2pJingleSession) {
+            allPromises.push(
+                this.p2pJingleSession.replaceTrack(oldTrack, newTrack));
+        }
+        return Promise.all(allPromises);
+    }
+
+    /**
+     * Intercepts events emitted by parent <tt>JitsiConference</tt>
+     * @private
+     */
+    _emitIntercept(eventType) {
+        const shouldBlock = this._shouldBlockEvent(eventType);
+        switch (eventType) {
+            // Log events which may be of interest for the P2P implementation
+            case JitsiConferenceEvents.CONNECTION_INTERRUPTED:
+            case JitsiConferenceEvents.CONNECTION_RESTORED:
+            case JitsiConferenceEvents.P2P_STATUS:
+                logger.debug(
+                    "_emitIntercept: block? " + shouldBlock, arguments);
+                break;
+        }
+        if (!shouldBlock) {
+            this._originalEmit.apply(this.eventEmitter, arguments);
```

Comment:
For me in this case it's easier to grab whole 'arguments' array and pass it to 'apply'. Otherwise I would have to concat 'eventType' for 'apply', no ? How would you write it ? I must be missing something